tests/load/c/ginkgo_test.go

```diff
@@ -64,30 +64,36 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
 
 var _ = ginkgo.Describe("[Load Simulator]", ginkgo.Ordered, func() {
 	var (
-		network  *tmpnet.Network
-		tracker  *load.Tracker[common.Hash]
-		logger   = logging.NewLogger("c-chain-load-testing", logging.NewWrappedCore(logging.Info, os.Stdout, logging.Auto.ConsoleEncoder()))
-		tc       = e2e.NewTestContext()
-		r        = require.New(tc)
-		registry = prometheus.NewRegistry()
-	)
+		network *tmpnet.Network
+		tc      *e2e.GinkgoTestContext
+		tracker *load.Tracker[common.Hash]
 
-	tracker, err := load.NewTracker[common.Hash](registry)
-	r.NoError(err)
+		logger = logging.NewLogger("c-chain-load-testing", logging.NewWrappedCore(logging.Info, os.Stdout, logging.Auto.ConsoleEncoder()))
+	)
 
 	ginkgo.BeforeAll(func() {
-		tc := e2e.NewTestContext()
-		r := require.New(tc)
-		env := e2e.GetEnv(tc)
+		var (
+			tc  = e2e.NewTestContext()
+			env = e2e.GetEnv(tc)
+		)
 
 		network = env.GetNetwork()
 		network.Nodes = network.Nodes[:nodesCount]
 		for _, node := range network.Nodes {
 			err := node.EnsureKeys()
-			r.NoError(err, "ensuring keys for node %s", node.NodeID)
+			require.NoError(tc, err, "ensuring keys for node %s", node.NodeID)
 		}
+	})
 
-		cleanup := setupMetricsServer(r, registry, network.UUID, network.Owner, logger)
+	// Setup metrics server and load tracker before each test run
+	// ensuring isolation
+	ginkgo.BeforeEach(func() {
+		promRegistry := prometheus.NewRegistry()
+		loadTracker, err := load.NewTracker[common.Hash](promRegistry)
+		require.NoError(tc, err)
+		tracker = loadTracker
+
+		cleanup := setupMetricsServer(tc, promRegistry, network.UUID, network.Owner, logger)
```

Comment:
What's preventing this from being in the execute function (to reduce coupling with ginkgo)? We could pass the network to the execute function I think?